client/src/components/Form.jsx

```diff
@@ -0,0 +1,268 @@
+      selectedNights: 0,
+      check_in: 0,
+      check_out: 0,
+      selectedDate: 0,
+      check_in_clicked: false,
+      check_out_clicked: false,
+    };
+    this.increaseAdults = this.increaseAdults.bind(this);
+    this.increaseChildren = this.increaseChildren.bind(this);
+    this.increaseInfants = this.increaseInfants.bind(this);
+    this.decreaseAdults = this.decreaseAdults.bind(this);
+    this.decreaseChildren = this.decreaseChildren.bind(this);
+    this.decreaseInfants = this.decreaseInfants.bind(this);
+    this.guestButtonMessage = this.guestButtonMessage.bind(this);
+    this.guestSelectToggle = this.guestSelectToggle.bind(this);
+    this.guestExpandToggle = this.guestExpandToggle.bind(this);
+    this.calculateCostPerNight = this.calculateCostPerNight.bind(this);
+    this.onDayClick = this.onDayClick.bind(this);
+    this.handleCheckinClicked = this.handleCheckinClicked.bind(this);
+    this.handleCheckoutClicked = this.handleCheckoutClicked.bind(this);
+    this.calendarInitialize = this.calendarInitialize.bind(this);
+    this.updateTotalNights = this.updateTotalNights.bind(this);
+  }
+
+  onDayClick(e, dateContext, cb1, cb2) {
+    if (this.state.check_in_clicked) {
+      this.setState({
+        check_in: dateContext.format('MM/DD/YYYY'),
+      }, cb1());
+    } else if (this.state.check_out_clicked) {
+      this.setState({
+        check_out: dateContext.format('MM/DD/YYYY'),
+      }, cb2());
+    }
+  }
+
+  guestButtonMessage() {
+    if (this.state.adults === 1) {
+      this.setState({
+        adultMessage: '1 Guest',
+      });
+    } else {
+      this.setState({
+        adultMessage: `${this.state.adults} Guests`,
+      });
+    }
+
+    if (this.state.children === 1) {
+      this.setState({
+        childrenMessage: ', 1 Child',
+      });
+    } else {
+      this.setState({
+        childrenMessage: `, ${this.state.children} Children`,
+      });
+    }
+
+    if (this.state.infants === 1) {
+      this.setState({
+        infantMessage: ', 1 Infant',
+      });
+    } else {
+      this.setState({
+        infantMessage: `, ${this.state.infants} Infants`,
+      });
+    }
+  }
+
+  guestSelectToggle(e) {
+    this.setState({
+      guestSelected: true,
+    }, this.calculateCostPerNight);
+    e.preventDefault();
+  }
+
+  guestExpandToggle(e) {
+    this.setState({
+      guestExpand: !this.state.guestExpand,
+      guestSelected: true,
+    }, this.calculateCostPerNight);
+    this.updateTotalNights();
+    e.preventDefault();
+  }
+
+  calculateCostPerNight() {
+    let cost = this.props.price + this.props.service_fee + this.props.cleaning_fee;
+    let tax = cost * (this.props.tax / 100);
+    tax = parseFloat(tax.toFixed(2));
+    cost += tax;
+    cost = parseFloat(cost.toFixed(2));
+    const totalCost = cost * this.state.selectedNights;
+    this.setState({
+      totalCostPerNight: cost,
+      calculatedTax: tax,
+      totalCost,
+    });
+  }
+
+  increaseAdults(e) {
```

Comment:
I guess you have 3 html tags:
`
<tag className="adults"></tag>
<tag className="children"></tag>
<tag className="infants"></tag>
`
There is a simple way to use 1 function to achieve the functinos below:

`
 increaseCustomer(e) {
    e.preventDefault(e);
    
    this.setState({
        [e.target.className]: this.state.(e.target.className) + 1, //not sure if this.state.(e.target.className) is a correct statement.
    } , this.guestButtonMessage)
} 

` 
Instead of using 3 functions:
`  increaseAdults(e) {
    e.preventDefault();
    this.setState({
      adults: this.state.adults + 1,
    }, this.guestButtonMessage);
  }

   increaseChildren(e) {
    e.preventDefault();
    this.setState({
      children: this.state.children + 1,
    }, this.guestButtonMessage);
  }

   increaseInfants(e) {
    e.preventDefault();
    this.setState({
      infants: this.state.infants + 1,
    }, this.guestButtonMessage);
  }`